src/Locale.cpp

```diff
@@ -120,11 +120,26 @@ Locale::StringToCurrency(const char* string, Fixed& amount)
 	BNumberFormat numberFormatter;
 	BString decimalSymbol = numberFormatter.GetSeparator(B_DECIMAL_SEPARATOR);
 	BString groupingSeparator = numberFormatter.GetSeparator(B_GROUPING_SEPARATOR);
-
 	BString dollars(string), cents;
 
-	dollars.RemoveAll(fCurrencySymbol);
 	dollars.RemoveAll(groupingSeparator);
+	// Remove currency symbol
+	if (strcmp(fCurrencySymbol, "") == 0) {
+		// Remove currency symbol
+		// https://discuss.haiku-os.org/t/parsing-a-currency-formatted-string-to-number/15077/3
+		BString format5, format6;
+		numberFormatter.FormatMonetary(format5, 5.0);
+		numberFormatter.FormatMonetary(format6, 6.0);
+		int i = 0;
+		while (i < format5.Length() && format5[i] == format6[i])
+			i++;
+		while (i >= 0 && (format5[i] & 0xc0) == 0x80)
+			i--;
+		format5.Truncate(i);
+		dollars.RemoveFirst(format5);
+	} else {
```

Comment:
Remove opening curly and the closing one two lines down.

Reply:
Updated PR

Reply:
Thanks!